controllers/authors.js

```diff
@@ -0,0 +1,39 @@
+const models = require('../models')
+
+const getAllAuthors = async (request, response) => {
+  try {
+    const allAuthors = await models.authors.findAll()
+
+    return response.send(allAuthors)
+  } catch (error) {
+    return response.status(500).send('Could not retrieve authors, please try again.')
+  }
+}
+
+const getAuthorByIdWithNovelsAndGenres = async (request, response) => {
+  try {
+    const { id } = request.params
```

Comment:
This is no long just an id, it is an id or a name so a better name should be chosen like `identifier` or `searchTerm` or something like that